mino/httpclient/mod.go

```diff
@@ -0,0 +1,140 @@
+package httpclient
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"strings"
+	"time"
+)
+
+type key int
+
+const (
+	requestIDKey key = 0
+)
+
+// Httpclient defines the primitives to implement an http client that handles
+// client side requests
+type Httpclient interface {
+	// Start starts the http server. This call is assumed to be blocking
+	Start()
+
+	// RegisterHandler registers a new handler
+	RegisterHandler(path string, handler func(http.ResponseWriter, *http.Request)) error
+}
+
+// NewClient creates a new http client
+func NewClient(listenAddr string) *Client {
+	logger := log.New(os.Stdout, "httpclient :", log.LstdFlags)
```

Comment:
`logger := log.New(dela.Logger, "", 0)` to keep consistent logging.

Reply:
https://github.com/rs/zerolog#integration-with-nethttp for out of the box support.